File: src/main/java/io/github/addoncommunity/galactifun/api/universe/Galaxy.java
package io.github.addoncommunity.galactifun.api.universe;

import org.bukkit.inventory.ItemStack;

import io.github.addoncommunity.galactifun.api.universe.attributes.Orbit;
import io.github.addoncommunity.galactifun.api.universe.types.GalaxyType;

/**
 * A galaxy filled with star systems
 *
 * @author Mooy1
 */
public final class Galaxy extends UniversalObject {

    public Galaxy(String name, String id, GalaxyType type, Orbit orbit, TheUniverse orbiting, ItemStack baseItem) {
        super(name, id, type, orbit, orbiting, baseItem);
    }

}

File: src/main/kotlin/io/github/addoncommunity/galactifun/api/objects/Galaxy.kt
package io.github.addoncommunity.galactifun.api.objects

import io.github.addoncommunity.galactifun.api.objects.properties.Orbit
import org.bukkit.Material
import org.bukkit.inventory.ItemStack

class Galaxy(name: String, baseItem: ItemStack, orbiting: UniversalObject, orbit: Orbit) :
    UniversalObject(name, baseItem, orbiting, orbit) {

    constructor(name: String, baseItem: Material, orbiting: UniversalObject, orbit: Orbit) : this(
        name,
        ItemStack(baseItem),
        orbiting,
        orbit
    )
}